File: 백준/Silver/9461. 파도반 수열/파도반 수열.js
const fs = require('fs')

const [tc, ...arr] = fs.readFileSync('dev/stdin').toString().trim().split("\n").map(Number);

const dp = [0, 1, 1, 1, 2, 2, 3, 4, 5, 7, 9]

const solve = (n) => {
    for(let i = dp.length; i <= n; i++){
        dp.push(dp.at(-1) + dp.at(-5));
    }
    
    return dp[n];
}

for(let a of arr){
    console.log(solve(a))
}

File: 백준/Silver/9461. 파도반 수열/파도반 수열.py
import sys

input = sys.stdin.readline

tc= int(input())

arr = [int(input()) for _ in range(tc)]

dp = [0, 1, 1, 1, 2, 2, 3, 4, 5, 7, 9]

def solve(n):
    for i in range(len(dp), n + 1):
        dp.append(dp[-1] + dp[-5])
        
    return dp[n]
        

def main():
    for a in arr:
        print(solve(a))

main()
    